src/stores/wordpress-versions-api.ts

```diff
@@ -0,0 +1,167 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import * as Sentry from '@sentry/electron/renderer';
+import { __, sprintf } from '@wordpress/i18n';
+import { z } from 'zod';
+import { isWordPressDevVersion, isWordPressBetaVersion } from 'src/lib/wordpress-version-utils';
+import { withOfflineCheck } from 'src/stores/tests/utils/with-offline-check';
+
+const MINIMUM_WORDPRESS_VERSION = '5.9.9';
+
+const wordPressOfferSchema = z.object( {
+	version: z.string(),
+	response: z.string(),
+} );
+
+type WordPressOffer = z.infer< typeof wordPressOfferSchema >;
+
+const wordPressApiResponseSchema = z.object( {
+	offers: z.array( z.any() ),
+} );
+
+type WordPressApiOffer = {
+	version: string;
+	response?: string;
+	[ key: string ]: unknown;
+};
+
+type ProcessedOffer = {
+	version: string;
+	shortName: string;
+};
+
+const extractShortName = ( version: string ): string => {
+	if ( isWordPressDevVersion( version ) ) {
+		return 'nightly';
+	}
+	const match = version.match( /^(\d+\.\d+)/ );
+	return match ? match[ 1 ] : version;
+};
+
+export interface WordPressVersion {
+	isBeta: boolean;
+	isDevelopment: boolean;
+	isLatest: boolean;
+	label: string;
+	value: string;
+}
```

Comment:
Nitpicky, but I would move `WordPressOffer`, `WordPressApiOffer`, `ProcessedOffer`, and `WordPressVersion` closer to where they're used.